dipy/nn/utils.py

```diff
@@ -80,45 +81,71 @@ def set_logger_level(log_level, logger):
+    voxsize=(1, 1, 1),
+    target_voxsize=None,
     considered_points="corners",
-    init_shape=(256, 256, 256),
-    scale=2,
+    init_shape=None,
+    ratio=None,
+    final_size=None,
+    need_isotropic=False,
+    order=3,
 ):
     """
-    Function to reshape image as an input to the model
+    Function to transform images for Deep Learning models
 
     Parameters
     ----------
     image : np.ndarray
-        Image to transform to voxelspace
+        Image to transform.
     affine : np.ndarray
-        Affine matrix provided by the file
-    voxsize : np.ndarray (3,), optional
-        Voxel size of the image
+        Affine matrix provided by the image file.
+    voxsize : tuple (3,), optional
+        Voxel size provided by the image file.
+    target_voxsize : tuple (3,), optional
+        The voxel size we want to start from.
+        Ignored if need_isotropic is True.
     considered_points : str, optional
-        which points to consider when calculating
-        the boundary of the image. If there is shearing
-        in the affine, 'all' might be more accurate
-    init_shape : list, tuple or numpy array (3,), optional
-        Initial shape to transform the image to
-    scale : float, optional
-        How much we want to scale the image
+        Considered points when calculating the transformed shape.
+        \"corners\" will consider only corners of the image shape.
+        \"all\" will consider all voxels. Might be needed when shearing is applied.
+    init_shape : tuple (3,), optional
+        What we want the initial shape to be before last resizing step.
+        Ignored if need_isotropic is True.
+    ratio : float, optional
+        The ratio of change in the last resizing step.
+        Ignored if need_isotropic is True.
+    final_size : tuple (3,), optional
+        The final size of the image array.
+    need_isotropic : bool, optional
+        Whether the output needs to be isotropic in the end.
+    order : int, optional
+        The order of the spline interpolation.
+        The order has to be in the range 0-5.
+        If transforming an int image, order 0 is recommended.
+
+
 
     Returns
     -------
-    tuple
-        Tuple with variables for recover_img
+    new_image : np.ndarray
+        Transformed image to be used in the Deep Learning model.
+    params : tuple
+        Parameters that are used when recovering the original image space.
     """
-    if voxsize is not None and np.any(voxsize != np.ones(3)):
-        image2, affine2 = reslice(image, affine, voxsize, (1, 1, 1))
+    ori_shape = image.shape
+    if need_isotropic:
+        target_voxsize = tuple(np.max(voxsize) * np.ones(3))
+    if target_voxsize is not None and np.any(target_voxsize != np.ones(3)):
+        image2, affine2 = reslice(image, affine, voxsize, target_voxsize)
     else:
         image2 = image.copy()
         affine2 = affine.copy()
+
     shape = image2.shape
 
     if considered_points == "corners":
```

Comment:
Let's create a method that only focuses on returning corners. and consider better naming convention for the inner variables.